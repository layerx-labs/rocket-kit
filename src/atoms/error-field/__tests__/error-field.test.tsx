import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { ErrorField } from '../../..';

describe('ErrorField', () => {
  it('renders', () => {
    const { asFragment } = render(<ErrorField error={'operation failed.'} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct error text', () => {
    const fieldText = 'Failed to save name';
    render(<ErrorField error={fieldText} color="danger" />);
    screen.getByText(fieldText);
  });

  it('has correct success text', () => {
    const fieldText = 'Name save!';
    render(<ErrorField error={fieldText} color="success" />);
    screen.getByText(fieldText);
  });
});
