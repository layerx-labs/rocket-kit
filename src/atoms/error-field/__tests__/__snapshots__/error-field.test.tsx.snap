// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorField renders 1`] = `
<DocumentFragment>
  .c0 {
  display: block;
  font-size: 0.7rem;
  color: hsl(354,83%,64%);
}

<span
    class="c0 error-field"
    color="danger"
  >
    operation failed.
  </span>
</DocumentFragment>
`;
